### PR TITLE
Track kibana system role in CODEOWNERS

### PR DESCRIPTION
**Pull request by n1v0lg**
The ES security team currently manually tags the Kibana security team for reviews on changes to the Kibana system role definition. This PR aims to automate via codeowners.

### CODE REVIEW

**Comment by n1v0lg** [comment]
@elastic/es-delivery I think I need to grant write access to the relevant Kibana team (`@elastic/kibana-security`) for this to work. Getting:

`Unknown owner on line 24: make sure the team @elastic/kibana-security exists, is publicly visible, and has write access to the repository` otherwise

Could you validate that we're good with this? I think I have access to do this, just want to make sure I get a +1 from delivery before I proceed.

**Review by ywangd — APPROVED** [review]
LGTM

There is also a `kibana_admin` role. But I am OK with *not* moving it since it hardly changes. We can get to it when it actually becomes a problem.

> I think I need to grant write access to the relevant Kibana team

I am not sure this is indeed necessary. The same error shows for `@elastic/fleet` team which is an existing entry in the file. What we need from this file is mostly just the notification function. So it should work even without write permission?

**Comment by n1v0lg** [comment]
> There is also a kibana_admin role. But I am OK with not moving it since it hardly changes. We can get to it when it actually becomes a problem.

I'll move it for consistency -- true it doesn't change much but the only downside of moving it is an increase in method visibility from private to package-private which I think is worth the consistency gain.

> I am not sure this is indeed necessary.

I don't think it works without it (briefly looked at an open feature request: https://github.com/orgs/community/discussions/23042 and the docs: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners)

Based on the convo around the PR that added Fleet to code owners: https://github.com/elastic/elasticsearch/pull/91670 I think we did at some point give the fleet team write permissions (and then must have removed them again?). The codeowners file specifies that fleet should get notified on `x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet` but a recent [PR](https://github.com/elastic/elasticsearch/pull/97306) that touched `x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/Fleet.java` did not result in a notification or review request.

File: .github/CODEOWNERS
Comment (azasypkin) [inline .github/CODEOWNERS:24]: __question:__ Does it make sense to add your team as a co-owner as well? I suppose currently you are only mentioned if specific labels are added. I'm not sure if you want to change that, though.

Comment (n1v0lg) [reply .github/CODEOWNERS:24]: Good q! I think we should stick with the existing process for now, since it's already established and working. We can always re-assess later.

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Pull request merged**
